repro/Proxy.cxx

```diff
@@ -462,8 +462,24 @@ Proxy::thread()
                }
                else
                {
-                  InfoLog (<< "No matching request context...ignoring " << *app);
-                  delete app;
+                   RequestContextMap::iterator i = mClientRequestContexts.find(tid);
+                   if (i != mServerRequestContexts.end())
```

Comment:
You are doing a find on mClientRequestContexts on line 465, but checking mServerRequestContexts.end() on 466.

Reply:
Well spotted, thanks!